Fix for Studio Accessories on scene load

Vaaskr opened the pull request:
This should prevent accessories from being set to always active on when a scene is loaded even when an accessory state was set to off when saving a scene

Vaaskr committed:
Fix Studio Accessories not properly setting on scene load

starstormhun commented on src/AmazingNewAccessoryLogic/AmazingNewAccessoryLogic/ANAL.CharacterController.cs:
```diff
@@ -1253,15 +1253,18 @@ protected override void Update()
 
                 IEnumerator UpdateLater()
                 {
-                    for (int i = 0; i < 2; i++) yield return null;
```
The yields need to remain at the start here.

starstormhun commented on src/AmazingNewAccessoryLogic/AmazingNewAccessoryLogic/ANAL.CharacterController.cs:
```diff
@@ -1253,15 +1253,18 @@ protected override void Update()
-                        for (int i = 0; i < ChaControl.infoAccessory.Length; i++)
-                            if (ChaControl.infoAccessory[i] != null)
-                                setAccessoryState(i, true);
+                        for (int i = 0; i < 2; i++) yield return null;
```
Remove the yields from here.

starstormhun commented on src/AmazingNewAccessoryLogic/AmazingNewAccessoryLogic/ANAL.CharacterController.cs:
```diff
@@ -1253,15 +1253,18 @@ protected override void Update()
                 {
-                    for (int i = 0; i < 2; i++) yield return null;
-                    if (lastCoordHadANAL)
+                    if((!StudioAPI.InsideStudio)
```
There's no need for two nested ifs, just place the two conditions into a single if with &&

starstormhun reviewed (changes requested):
Needs some changes still.

Vaaskr committed:
Update ANAL.CharacterController.cs

Thanks to @Starstorm for some advice for a better way to structure the studio fix

starstormhun reviewed (approved):
Looks good now.
